### PR TITLE
SWC-6657

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/widget/profile/UserProfileEditorWidgetImpl.java [hunk 1]
```diff
@@ -120,28 +120,33 @@ public void configure(
...
       // find out what the primary (notification) email address is
       // SWC-5599: the first email is not the notification email address.
-      jsClient.getNotificationEmail(
-        new AsyncCallback<NotificationEmail>() {
-          @Override
-          public void onFailure(Throwable caught) {
-            synAlert.handleException(caught);
-          }
+      if (isCurrentUserProfile) {
+        jsClient.getNotificationEmail(
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/profile/UserProfileEditorWidgetImpl.java:131]: Only get the Notification email if the user is looking at their own profile

File: src/main/java/org/sagebionetworks/web/client/widget/profile/UserProfileEditorWidgetViewImpl.java [hunk 2]
```diff
@@ -213,8 +215,29 @@ public void setUsername(String username) {
   }
 
   @Override
-  public void setEmail(String email) {
-    this.email.setText(email);
+  public void setEmails(List<String> emails, String notificationEmail) {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/profile/UserProfileEditorWidgetViewImpl.java]: Changed the email textbox to a div container, and we just add paragraph elements (with tooltips) to it.
If it's the notification email, list it first and in bold.

File: src/test/java/org/sagebionetworks/web/unitclient/widget/profile/UserProfileEditorWidgetImplTest.java [hunk 3]
```diff
@@ -214,7 +214,6 @@ private void verifyConfigure(int nCalls) {
     verify(mockView, times(nCalls)).setLocation(profile.getLocation());
     verify(mockView, times(nCalls)).setLink(profile.getUrl());
     verify(mockView, times(nCalls)).setBio(profile.getSummary());
-    verify(mockView, times(nCalls)).setEmail(EMAIL2);
```
Comment (jay-hodgson) [inline src/test/java/org/sagebionetworks/web/unitclient/widget/profile/UserProfileEditorWidgetImplTest.java:217]: verifyConfigure() no longer checks for view.setEmail(), we do this in the individual test scenarios.

File: src/test/java/org/sagebionetworks/web/unitclient/widget/profile/UserProfileEditorWidgetImplTest.java [hunk 4]
```diff
@@ -225,6 +224,7 @@ public void testConfigure() {
     widget.configure(profile, ORC_ID, mockCallback);
 
     verifyConfigure(1);
+    verify(mockView).setEmails(userEmails, EMAIL2);
```
Comment (jay-hodgson) [inline src/test/java/org/sagebionetworks/web/unitclient/widget/profile/UserProfileEditorWidgetImplTest.java:228]: For a normal configure, where user is looking at their own profile, view.setEmails() should be called with both the array of emails and notification email.

File: src/test/java/org/sagebionetworks/web/unitclient/widget/profile/UserProfileEditorWidgetImplTest.java [hunk 5]
```diff
@@ -239,9 +239,23 @@ public void testConfigureNoEmail() {
   @Test
   public void testConfigureNotOwner() {
     when(mockAuthController.getCurrentUserPrincipalId()).thenReturn(null); //anonymous
```
Comment (jay-hodgson) [inline src/test/java/org/sagebionetworks/web/unitclient/widget/profile/UserProfileEditorWidgetImplTest.java:242]: If not the owner and not ACT, then emails should not be set (by the service), and view.setEmails() should not be called.

File: src/test/java/org/sagebionetworks/web/unitclient/widget/profile/UserProfileEditorWidgetImplTest.java [hunk 6]
```diff
@@ -239,9 +239,23 @@ public void testConfigureNoEmail() {
...
+    profile.setEmails(null);
+    widget.configure(profile, ORC_ID, mockCallback);
+
+    verifyConfigure(1);
+    verify(mockView, never()).setEmails(anyList(), anyString());
+    verify(mockView).setCanEdit(false);
+  }
+
+  @Test
+  public void testConfigureACT() {
```
Comment (jay-hodgson) [inline src/test/java/org/sagebionetworks/web/unitclient/widget/profile/UserProfileEditorWidgetImplTest.java:252]: If ACT, then the emails should now be returned, but we are unable to get the notification email.

File: src/test/java/org/sagebionetworks/web/unitclient/widget/profile/UserProfileEditorWidgetImplTest.java [hunk 7]
```diff
@@ -203,6 +203,7 @@ private void verifyConfigure(int nCalls) {
     verify(mockSynAlert, times(nCalls)).clear();
     verify(mockView, times(nCalls)).hideLinkError();
     verify(mockView, times(nCalls)).hideUsernameError();
+    verify(mockView, times(nCalls)).clearEmails();
```
Comment (jay-hodgson) [inline src/test/java/org/sagebionetworks/web/unitclient/widget/profile/UserProfileEditorWidgetImplTest.java:206]: Add a new clearEmails() method, and verify that it is called whenever the profile editor is configured

**Review by nickgros — APPROVED** [review]

**Pull request merged**